src/mono/browser/runtime/marshal-to-js.ts

```diff
@@ -348,11 +349,17 @@ export function mono_wasm_resolve_or_reject_promise(args: JSMarshalerArguments):
         mono_assert(holder, () => `Cannot find Promise for JSHandle ${js_handle}`);
 
         holder.resolve_or_reject(type, js_handle, arg_value);
-
+        if (MonoWasmThreads) {
+            Module._free(args as any);
```

Comment:
Is this meant to pair with an AllocHGlobal in managed code? I don't love the idea of implicitly assuming that AllocHGlobal is Module._malloc and not some other allocator

Reply:
we make that assumption in many places already. I added comment.